### PR TITLE
[FEAT] Implement cd builtin function

### PR DESCRIPTION
**Pull request by itislu**
I tried to implement cd exactly as described in the 10 steps in the POSIX specification for cd:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cd.html

Also, @LeaYeh please help me with the exit codes, I am a bit lost which ones I can use now.
If a critical error occurs, like a `malloc()` failure, should I return `SUBSHELL_ERROR`?

### CODE REVIEW

File: source/utils/final_cmd_table_utils.c
Comment (itislu) [inline source/utils/final_cmd_table_utils.c:1]: This is probably one of the important changes you should know about.

Commit message:
```
fix: Fix potential segfault in setup_env
A segfault could happen if the key and the value of an env node were longer than PATH_MAX.
Now, the exact size gets malloced.
```

File: libraries/libft/src/libft/strings/ft_strdel_sequence.c
Comment (itislu) [inline libraries/libft/src/libft/strings/ft_strdel_sequence.c:1]: New libft function, just so you know about its existance.

File: libraries/libft/src/libft/lists/singly_linked/ft_lstdrop_node.c
Comment (itislu) [inline libraries/libft/src/libft/lists/singly_linked/ft_lstdrop_node.c:1]: New libft function, just so you know about its existance.

File: libraries/libft/src/libft/strings/ft_rplc_part_of_str.c
Comment (itislu) [inline libraries/libft/src/libft/strings/ft_rplc_part_of_str.c:1]: New libft function, just so you know about its existance.

File: libraries/libft/src/libft/strings/ft_strrplc_sequence.c
Comment (itislu) [inline libraries/libft/src/libft/strings/ft_strrplc_sequence.c:1]: New libft function, just so you know about its existance.

**Review by LeaYeh — APPROVED** [review]

**Pull request merged**